From baa5cd72f9a1bc1818fe34bccdf6d7eb5c73f43a Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Tue, 18 Feb 2025 13:41:28 +0800
Subject: [PATCH 001/119] mtk: mt76: mt7996: config rxfilter to drop other
 unicast by default

Set MT_WF_RFCR_DROP_OTHER_UC by default and disable this flag in
mt7996_set_monitor only if monitor mode is enabled.

Without this patch, the MT_WF_RFCR_DROP_OTHER_UC would not be set so the
driver would receive lots of packets that is not sent to me.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7996/init.c b/mt7996/init.c
index a53d8ee2..f9ee8969 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -402,6 +402,7 @@ mt7996_init_wiphy_band(struct ieee80211_hw *hw, struct mt7996_phy *phy)
 
 	phy->slottime = 9;
 	phy->beacon_rate = -1;
+	phy->rxfilter = MT_WF_RFCR_DROP_OTHER_UC;
 
 	if (phy->mt76->cap.has_2ghz) {
 		phy->mt76->sband_2g.sband.ht_cap.cap |=
-- 
2.45.2

